altos-rust/altos-core/src/task/control.rs

```diff
@@ -63,6 +69,13 @@ impl Priority {
     (0..NUM_PRIORITIES)
   }
 
+
+  /// Returns a box of the range of values of Priority minus the one passed in.
+  pub fn all_except<'a>(priority:Priority) -> Box<Iterator<Item=usize> + 'a> {
+    let all = Priority::all();
+    Box::new(all.filter(move |&x| x != priority as usize))
```

Comment:
We don't want to be doing allocations every time the scheduler runs, this value is going to be used then dropped almost immediately, which could cause some issues with memory fragmentation. Unfortunately we've got to do some strange stuff to keep the type checker happy, but you can just return the filtered sequence directly:
```rust
pub fn all_except<P>(predicate: P) -> Filter<Range<usize>, P> where P: Fn(&usize) -> bool {
    Priority::all().filter(predicate)
}
```

Then at the calling site you'd say something like 
```rust
let queues = Priority::all_except(|&i| i != Priority::Normal as usize);
```

It's a bit quirky and we'd probably want to come back and examine both the `Priority::all()` and `Priority::all_except()` functions in the future to see if we can clean them up, but it's better than doing the allocations.

We may just want to write our own iterator type and return that from these two functions instead.